File: llm_written_v1/gemini-2.5-pro/bfs/go/graph.go
package llm_written

// Graph represents an undirected graph using an adjacency list.
type Graph struct {
	Vertices map[int][]int
}

// AddEdge adds an undirected edge between two vertices.
func (g *Graph) AddEdge(from, to int) {
	if g.Vertices == nil {
		g.Vertices = make(map[int][]int)
	}
	g.Vertices[from] = append(g.Vertices[from], to)
	g.Vertices[to] = append(g.Vertices[to], from)
}

File: llm_written/claude-sonnet-4-5/bfs/cpp/graph.cpp
#include "graph.h"

void Graph::add_edge(int u, int v) {
    if(vertices.count(u) == 0) vertices[u] = {};
    if(vertices.count(v) == 0) vertices[v] = {};

    vertices[u].push_back(v);
    vertices[v].push_back(u);
}
